docs/spec_draft.md

```diff
@@ -339,6 +349,170 @@ only exists to establish data dependencies from `result` to `inputs`.
+instance (`ei`) of `rid`, `pid` pair, henceforth referred to as `ei(rid, pid)`
+into groups. All the execution instances within a group can participate in the
+operation without getting interfered by execution instances from other groups.
+The number of groups represents the number of `stablehlo.all_to_all` operations
+executed in parallel.
+
+Given,
+  * `num_pids`: Number of partition ids
+  * `num_rids`: Number of replica ids
+  * `all_partions_ids`: { pid : pid $\in$ [0, `num_pids`) }
+  * `all_replica_ids`: { rid : rid $\in$ [0, `num_rids`) }
+  * `size_subgroup`: size of each subgroup in `replica_groups`.
+  * `num_subgroup`: number of subgroups in `replica_groups`.
+
+The formation of groups is defined as follows:
+   * Number of groups formed: `num_subgroup` * `num_pids`.
+
+   * Size of each group: `size_subgroup`.
+
+   * The formation of `groups` can be demonstrated, using Python-like syntax, as
+     follows:
+     ```python
+     all_replica_ids = [rid for rid in range(num_rids)]
+     all_partition_ids = [pid for pid in range(num_pids)]
+     groups = []
+
+     if len(replica_groups) == 0:
+       replica_groups = [all_replica_ids]
+
+     for replica_group in replica_groups:
+       for pid in all_partition_ids:
+         sub_group = []
+         for rid in replica_group:
+           sub_group.append(ei(rid, pid))
+         groups.append(sub_group)
+     return groups
+     ```
+
+  * For example, assuming `num_pids = 2` and
+      `replica_groups = [[0, 2], [1, 3]]`:
+      `groups` formed: `[ei(0, 0), ei(2, 0)], [ei(0, 1), ei(2, 1)], [ei(1, 0), ei(3, 0)], and [ei(1, 1), ei(3, 1)]`.
+
+For each group `G` $\in$ `groups`, the operation can be described as:
+  * Each `operand`, corresponding to an execution instance in `G`,
+    is split into `split_count` number of blocks, described below using
+    Python-like syntax.
+    ```python
+    blocks = [
+        slice(
+          operand=operand,
+          start_indices=[s0, s1, ..., sR-1],
+            # where
+            #  - sk = 0 if k != split_dimension
+            #  - sk = j* dim(operand) / split_count, if k == split_dimension
+            #  - R = rank(operand)
+        limit_indices=[l0, l1, ..., lR-1],
+            # where
+            #   - lk = dim(operand, k)  if k != split_dimension
+            #   - lk = dim(operand, split_dimension) / split_count, if k == split_dimension
+        strides=[t0, t1, ..., tR-1],
+            # where
+            #   - lk = 0  if k != split_dimension
+            #   - lk = dim(operand, split_dimension) / split_count, if k == split_dimension
+        ) for j in range(0, split_count)]
+    ```
+  * `blocks` are scattered to all the execution instances in `G` such that
+    `blocks[i]` is send to ith execution instance, for i $\in$ G.
+  * Each execution instance in `G` concatenates the received
+    blocks along the `concat_dimension` to produce `result` which is given by
+    ```python
```

Comment:
Fixing some typos, and suggest to move description in comments:
```
  * Each execution instance in `G` concatenates the received blocks along the
    `concat_dimension` to produce `result` which is given by
    ```python
    # received_blocks = { received blocks sorted based on the order of
    # execution instances in G }
    result = concatenate(received_blocks, concat_dimension)
    ```
```

Reply:
I agree that adding description in code might be the right choice. 

How about this?

is given by
    `result = concatenate(received_blocks, concat_dimension)`,
        where `received_blocks = { received blocks sorted based on the order of
          execution instances in G }`.

based on other ops using similar style.

Reply:
Typos are fixed as requested.

Reply:
Can you also align the text to 80 char limit in this bullet